health/health.go

```diff
@@ -120,13 +120,29 @@ const (
 	SysTKA = Subsystem("tailnet-lock")
 )
 
-// NewWarnable returns a new warnable item that the caller can mark as health or
-// in warning state via Tracker.SetWarnable.
+var subsystemsWarnables = map[Subsystem]*Warnable{}
+
+// Warnable() returns a Warnable representing a legacy Subsystem. This is used
+// *temporarily* while we migrate the old health infrastructure based on
+// Subsystems to the new Warnables architecture.
+func (s Subsystem) Warnable() *Warnable {
+	if w, ok := subsystemsWarnables[s]; ok {
+		return w
+	} else {
+		w := NewWarnable(WarnableCode(fmt.Sprintf("legacy_%s", s)))
+		subsystemsWarnables[s] = w
+		return w
+	}
+}
+
+// NewWarnable returns a new warnable item that the caller can mark as healthy or
+// in warning state via Tracker.AddWarning.
 //
 // NewWarnable is generally called in init and stored in a package global. It
 // can be used by multiple Trackers.
-func NewWarnable(opts ...WarnableOpt) *Warnable {
+func NewWarnable(code WarnableCode, opts ...WarnableOpt) *Warnable {
 	w := new(Warnable)
+	w.Code = code
```

Comment:
keep a package-level map/set to make sure all registered warnables have a globally unique code.

Reply:
done